Added image upload functionality and changed name of index.html file

LilCSharp opened the pull request:
### What is a quick description of the change?

I added a JSP file and two Java servlets that have the ability to permit the retrieval of images uploaded to a test database

### Screenshot/Screengrab after this change?

![image](https://user-images.githubusercontent.com/50273061/86281597-d07ee400-bbab-11ea-93e3-9ccac48af523.png)


### Is this fixing an issue?

I changed the name of the index.html file so that Alicia can use index.html for development

### Are there more details that are relevant?

none

### Any additional comments?

This must be completely refactored when integration with the front-end occurs.

LilCSharp committed:
Preliminary backend media transfer functionality. DO NOT treat as final version. Will very likely be replaced

LilCSharp committed:
Created a basic image upload function that can be integrated into other systems

LilCSharp committed:
Created a basic image upload function that can be integrated into other systems, changed discrepencies

LilCSharp committed:
Updating to rebase

wu-haotian reviewed (commented):
I mentioned this in a previous PR as well... looks like there're multiple copies of the code, do you know why's that the case?

e.g. there are 4 imgHandling.js and they looks similar:
target/appengine-staging/js/imgHandling.js
target/step13-2020-1/js/imgHandling.js
target/appengine-staging/__static__/js/imgHandling.js
src/main/webapp/js/imgHandling.js

LilCSharp commented:
These are not copies. These are files the mvn needs and automatically creates when running the dev server and when deploying the site. Everything with target after it is typically auto-generated

wu-haotian commented:
> These are not copies. These are files the mvn needs and automatically creates when running the dev server and when deploying the site. Everything with target after it is typically auto-generated

I see, thanks for letting me know! If that's the case, I wonder if we should drop tracking those files with git. It makes code review more difficult, and may cause merge conflict in the future. @antrromet what's your suggestion here?

jamierashley commented:
Code for the servlet looks good to me. You can delete the target folder before committing I believe, It's not necessary but it makes the code a little confusing.

aliciasimon commented:
LGTM. The only thing I would say is to add some comments to your code so that people reviewing it can have a better understanding of what you've written.

wu-haotian commented on src/main/java/com/google/sps/servlets/ServeImgServlet.java:
```diff
@@ -0,0 +1,78 @@
+      }
+
+
+      /*
```
nit: either remove commented-out code, or add a comment on when to put the code back (same for the commented-out codes in other files)

e.g. // TODO: un-comment the code block after XXX.

wu-haotian commented on src/main/java/com/google/sps/servlets/ServeImgServlet.java:
```diff
@@ -0,0 +1,78 @@
+        BlobKey blobKey = new BlobKey(key);
+        blobstoreService.serve(blobKey, res);
+      }
+
```
nit: remove an extra empty line here

wu-haotian commented on src/main/java/com/google/sps/servlets/ServeImgServlet.java:
```diff
@@ -0,0 +1,78 @@
+import javax.servlet.annotation.WebServlet;
+
+@WebServlet("/serve")
+public class ServeImgServlet extends HttpServlet {
```
add a javadoc to explain what does this class do.

wu-haotian commented on src/main/java/com/google/sps/servlets/UploadImgServlet.java:
```diff
@@ -0,0 +1,110 @@
+import com.google.appengine.api.images.ServingUrlOptions;
+
+@WebServlet("/upload")
+public class UploadImgServlet extends HttpServlet {
```
add a javadoc to explain what does this class do.

wu-haotian commented on src/main/java/com/google/sps/servlets/UploadImgServlet.java:
```diff
@@ -0,0 +1,110 @@
+        
+        res.sendRedirect("/Pages/index.jsp");
+
+        /*
```
nit: remove or add comment for commented-out code blocks.

wu-haotian commented on src/main/webapp/js/imgHandling.js:
```diff
@@ -44,6 +36,7 @@ function handleFiles() {
       display.onload = sketch;
       display.src = URL.createObjectURL(this.files[i]);
 
+      /*
```
nit: remove or add comment for commented-out code blocks.

wu-haotian commented:
I synced with Antriksh on the auto-generated code issue. Antriksh said that we'll need a gitignore file to do that. I've filed https://github.com/googleinterns/step13-2020/issues/17 for this, we can also discuss in team chat

It's okay to do that (drop tracking on auto-generated files/directories) in a follow up PR, but please still fix other issues I commented in previous code reviews.

LilCSharp committed:
Added comments and image analysis functionality

LilCSharp commented:
I added comments and documentation. By the time that the review was done, I also added some additional stuff to the file, but most of it is commented out, so there should not be much of a change in functionality

LilCSharp committed:
More comments

wu-haotian commented:
> I added comments and documentation. By the time that the review was done, I also added some additional stuff to the file, but most of it is commented out, so there should not be much of a change in functionality
